File: Belot/Belot.Client/src/app/server-api/requests/register-request.ts
import { autoserializeAs } from "cerialize";
import BaseRequest from "./base-request";

export default class RegisterRequest extends BaseRequest {
  @autoserializeAs(String)
  username!: string;

  @autoserializeAs(String)
  email!: string;

  @autoserializeAs(String)
  password!: string;
}

File: Belot/Belot.Server/Models/Http/Requests/RegisterRequest.cs
namespace Belot.Models.Http.Requests
{
    using System.Runtime.Serialization;

    [DataContract]
    public class RegisterRequest : BaseRequest
    {
        [DataMember]
        // Other classes do not have this attribute, what's the idea ?
        public string Username { get; set; }

        [DataMember]
        public string Password { get; set; }

        [DataMember]
        public string Email { get; set; }
    }
}
